tracking/translations_parser/cli/taskcluster_group.py

```diff
@@ -95,109 +104,186 @@ def get_metrics_from_task(task: dict) -> list[Metric]:
-            [get_metrics_from_task(metrics_tasks.pop(task_id)) for task_id in dependent_tasks],
+            [get_metrics_from_task(metrics_tasks.pop(dependent_task_id)) for dependent_task_id in dependent_tasks],
             start=[],
         )
-        run = task["name"]
         publish_task(
-            project=project,
+            project=project_name,
             group=group_name,
-            name=run,
-            task=task,
+            name=training_task["name"],
+            task=training_task,
             metrics=metrics,
         )
 
-    # Group and publish missing runs with metrics via the logs publication
-    with tempfile.TemporaryDirectory() as d:
-        logs_folder = Path(d) / "logs"
-        eval_folder = logs_folder / project / group_name / "eval"
+    # Group and publish remaining metrics tasks via the logs publication
+    with tempfile.TemporaryDirectory() as temp_dir:
+        logs_folder = Path(temp_dir) / "logs"
+        eval_folder = logs_folder / project_name / group_name / "eval"
         eval_folder.mkdir(parents=True, exist_ok=True)
-        for task in metrics_tasks.values():
-            filename = task["task"]["tags"]["label"]
+
+        for metrics_task in metrics_tasks.values():
+            filename = metrics_task["task"]["tags"]["label"]
             # evaluate-teacher-flores-flores_aug-typos_devtest-lt-en-1/2
-            with (eval_folder / f"{filename}.log").open("wb") as f:
+            with (eval_folder / f"{filename}.log").open("wb") as log_file:
                 downloadArtifactToFile(
-                    f,
-                    taskId=task["status"]["taskId"],
+                    log_file,
+                    taskId=metrics_task["status"]["taskId"],
                     name="public/logs/live.log",
                     queueService=queue,
                 )
+
         # Dump experiment config so it is published on group_logs
-        config_path = Path(d) / "experiments" / project / group_name / "config.yml"
+        config_path = Path(temp_dir) / "experiments" / project_name / group_name / "config.yml"
         config_path.parent.mkdir(parents=True, exist_ok=True)
-        with config_path.open("w") as f:
-            yaml.dump(config, f)
+
+        with config_path.open("w") as config_file:
+            yaml.dump(config, config_file)
 
         parents = str(logs_folder.resolve()).strip().split("/")
-        WandB.publish_group_logs(parents, project, group_name, existing_runs=[], tag_sep="-")
+        WandB.publish_group_logs(parents, project_name, group_name, existing_runs=[], tag_sep="-")
+
+
+def list_dependent_group_ids(task_id: str, known: set[str]):
+    task = queue.task(task_id)
+
+    # Browse task dependencies
+    for dependent_task_id in task['dependencies']:
+        dependent_status = queue.status(dependent_task_id)
+
+        group_id = dependent_status["status"]["taskGroupId"]
+        if group_id in known:
+            continue
+
+        known.add(group_id)
```

Comment:
I think it is more explicit to yield each group, and iterate at above level adding elements to the set. \
Passing objects by references seems hard to follow reading code IMO.

Reply:
Fixed in 0df72095f51f41de3e8d9434797a66dea560289f